### PR TITLE
Improved: enhanced visibility of completed items in POs and displayed items count (#259)

### LINKED ISSUE #152: Show order and item status on POs
## What is the motivation for adding/enhancing this feature?

Only Approved POs can be received. Right now we don't show the status of the PO which makes it hard to know that can and cannot be received.

There should be a badge on the end slot of the find screen with a badge that shows the status.

On the detail page, next to the internal order id badge, there should be an order level status badge.

For items, the receive all button should be replaced with a "completed" badge when the item status has changed to completed.

### LINKED ISSUE #259: Enhanced Visibility of Completed Items in Purchase Orders (PO)
## What is the motivation for adding/enhancing this feature?
When a PO contains multiple items to be received, it's essential for users to have a clear understanding of which products have already been received or completed. The completed items will be clearly distinguishable from pending items on the Purchase Order detail page, ensuring that users can quickly identify which products are no longer outstanding. This feature is crucial for users dealing with large and complex POs, as it significantly improves their ability to manage and track received items.

## What are the acceptance criteria?

- [ ] ...

## Can you complete this feature request by yourself?

- [ ] YES
- [ ] NO

### PR DESCRIPTION
**Pull request by amansinghbais**
### Related Issues

Closes #259
Closes #152

### Short Description and Why It's Useful

Currently completed items on the Purchase Order detail page are not clearly distinguishable from pending items which doesn't allows users quickly identify which products are no longer outstanding. This feature is crucial for users dealing with large and complex POs. Hence,
- Written code to display completed items too on the POs detail page.
- Written a function and added few template items so as to clearly display total orders count, pending items count, and completed items count.

### Screenshots of Visual Changes before/after (If There Are Any)

Before
Screenshot from 2023-09-08 15-19-14

After
Screenshot from 2023-09-08 15-19-32

Screenshot from 2023-09-08 15-19-38

Screenshot from 2023-09-08 15-19-52

**IMPORTANT NOTICE** - Remember to add changelog entry

### Contribution and Currently Important Rules Acceptance

- [x] I read and followed [contribution rules](https://github.com/hotwax/receiving#contribution-guideline)

### CODE REVIEW

File: src/views/PurchaseOrderDetail.vue [hunk 1]
```diff
@@ -90,6 +98,44 @@
...
+            <div class="location">
+              <ion-chip :disabled="true" outline>
+                <ion-icon :icon="locationOutline"/>
+                <ion-label>{{ item.locationSeqId }}</ion-label>
+              </ion-chip>
+            </div>
+            
+            <div>
+              <ion-item lines="none" class="ion-float-end">
+                <ion-badge color="medium" class="ion-margin-end">{{ item.quantity }} {{ $t("ordered") }}</ion-badge>
```
Comment (k2maan) [inline src/views/PurchaseOrderDetail.vue:133]: Use slots instead of float styling and margin.

**Review by k2maan — CHANGES_REQUESTED** [review]

**Review by k2maan — APPROVED** [review]

File: src/views/PurchaseOrderDetail.vue [hunk 2]
```diff
@@ -204,6 +253,13 @@ export default defineComponent({
       if(this.queryString) payload = this.queryString
       this.store.dispatch('order/updateProductCount', payload)
     },
+    getPOItemsCount(orderType: string) {
+      if(orderType === 'completed') {
+        return this.order.items.filter((item: any) => item.orderItemStatusId === 'ITEM_COMPLETED').length
+      } else if(orderType === 'pending') {
```
Comment (ymaheshwari1) [inline src/views/PurchaseOrderDetail.vue:259]:
```suggestion
      } else {
```

We can remove this `if` as there is no other case we are handling for orderType.

File: src/views/PurchaseOrderDetail.vue [hunk 3]
```diff
@@ -90,6 +98,44 @@
             </div>         
           </div>
         </ion-card>
+
+        <ion-item lines="none">
+          <ion-text color="medium" class="ion-margin-end">
+            {{ $t("COMPLETED:", { itemsCount: getPOItemsCount('completed') > 1 ? `${getPOItemsCount('completed')} ${$t("ITEMS")}` : `${getPOItemsCount('completed')} ${$t("ITEM")}` }) }}
+          </ion-text>
```
Comment (ymaheshwari1) [inline src/views/PurchaseOrderDetail.vue:105]:
```suggestion
          <ion-text v-if="getPOItemsCount('completed') > 1" color="medium" class="ion-margin-end">
            {{ $t("COMPLETED: ITEMS", { itemsCount: getPOItemsCount('completed')}) }}
          </ion-text>
          <ion-text v-else color="medium" class="ion-margin-end">
            {{ $t("COMPLETED: ITEM", { itemsCount: getPOItemsCount('completed')}) }}
          </ion-text>
```

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

File: src/views/PurchaseOrderDetail.vue [hunk 4]
```diff
@@ -90,6 +98,44 @@
...
+          <ion-icon v-if="getPOItemsCount('completed') && showCompletedItems" :icon="eyeOutline" @click="showCompletedItems = !showCompletedItems" />
```
Comment (ymaheshwari1) [inline src/views/PurchaseOrderDetail.vue:106]:
```suggestion
          <ion-icon v-if="getPOItemsCount('completed')" :icon="showCompletedItems ? eyeOutline : eyeOffOutline" @click="showCompletedItems = !showCompletedItems" />
```

File: src/views/PurchaseOrderDetail.vue [hunk 5]
```diff
@@ -204,6 +253,20 @@ export default defineComponent({
       if(this.queryString) payload = this.queryString
       this.store.dispatch('order/updateProductCount', payload)
     },
+    getPOItemsCount(orderType: string) {
```
Comment (ymaheshwari1) [inline src/views/PurchaseOrderDetail.vue:256]: We can use a single method for getting the count and items and use `.length` in the template to get the count

**Review by ymaheshwari1 — APPROVED** [review]

**Pull request merged**